src/components/Slider.tsx

```diff
@@ -0,0 +1,153 @@
+      let scrollWidth: number =
+        carouselRef.current.scrollWidth - carouselRef.current.clientWidth;
+
+      arrowRightRef.current.style.display =
+        carouselRef.current.scrollLeft == scrollWidth ? "none" : "block";
+      arrowLeftRef.current.style.display =
+        carouselRef.current.scrollLeft == 0 ? "none" : "block";
+    }
+  };
+
+  const handleMouseDown = (
+    event: React.MouseEvent<HTMLDivElement, MouseEvent>
+  ) => {
+    setIsDragStart(true);
+    setPrevPageX(event.pageX); // horizontal coordinate of mouse click relative to the left edge of parent element
+    if (carouselRef.current) {
+      setPrevScrollLeft(carouselRef.current.scrollLeft); // current scroll starting from left of selected element
+    }
+  };
+
+  const handleMouseMove = (
+    event: React.MouseEvent<HTMLDivElement, MouseEvent>
+  ) => {
+    if (!carouselRef.current) return;
+    event.preventDefault();
+
+    // when we stop holding mouse button stop dragging
+    if (!isDragStart) {
+      return;
+    }
+
+    setIsDragging(true);
+    carouselRef.current.classList.add("dragging");
+
+    // scroll when moving mouse
+    setPositionDiff(event.pageX - prevPageX);
+    carouselRef.current.scrollLeft = prevScrollLeft - positionDiff;
+  };
+
+  const handleMouseUp = () => {
+    setIsDragStart(false);
+    if (carouselRef.current) carouselRef.current.classList.remove("dragging");
+    if (!isDragging) return;
+    setIsDragging(false);
+
+    // cleanup
+    setPositionDiff(0);
+  };
+
+  function handleTouchDown(event: React.TouchEvent<HTMLDivElement>) {
+    setIsDragStart(true);
+    setPrevPageX(event.touches[0].pageX);
+    if (carouselRef.current) {
+      setPrevScrollLeft(carouselRef.current.scrollLeft);
+    }
+  }
+
+  const handleTouchMove = (event: React.TouchEvent<HTMLDivElement>) => {
+    if (!carouselRef.current) return;
+    event.preventDefault();
+    if (!isDragStart) {
+      return;
+    }
+    setIsDragging(true);
+    carouselRef.current.classList.add("dragging");
+    setPositionDiff(event.touches[0].pageX - prevPageX);
+    carouselRef.current.scrollLeft = prevScrollLeft - positionDiff;
+  };
+
+  useLayoutEffect(() => {
+    showHideIcons();
+  });
+
+  return (
+    <div className="featured__right">
+      <div
+        ref={carouselRef}
+        className="featured__carousel"
+        onMouseDown={(e) => handleMouseDown(e)}
+        onMouseMove={(e) => handleMouseMove(e)}
+        onMouseUp={() => handleMouseUp()}
+        onMouseLeave={() => handleMouseUp()}
+        onTouchStart={(e) => handleTouchDown(e)}
+        onTouchMove={(e) => handleTouchMove(e)}
+        onTouchEnd={() => handleMouseUp()}
+      >
+        <span
+          id="left"
+          ref={arrowLeftRef}
+          onClick={() => {
+            if (carouselRef.current) carouselRef.current.scrollLeft -= 325;
+          }}
+        >
+          <img src={arrowLeft} alt="Arrow left" />
+        </span>
+        {value.map((v) => (
```

Comment:
Add unique id to each product and use it as value for `key` prop. Also use meaningful names, not `v`, `e` etc.